packages/login/src/components/ActivateUserForm/index.tsx

```diff
@@ -0,0 +1,300 @@
+  const { linkData } = useContext(ConfirmRouteContext);
+
+  const emailFromLink = linkData?.email ? linkData.email : "";
+
+  const [name, setName] = useState(linkData.firstname ?? "");
+  const [nameValid, setNameValid] = useState(true);
+  const [surName, setSurName] = useState(linkData.lastname ?? "");
+  const [surNameValid, setSurNameValid] = useState(true);
+  const [password, setPassword] = useState("");
+  const [passwordValid, setPasswordValid] = useState(true);
+  const [isPasswordErrorShow, setIsPasswordErrorShow] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const onChangeName = (e: ChangeEvent<HTMLInputElement>) => {
+    setName(e.target.value);
+    setNameValid(true);
+  };
+
+  const onChangeSurName = (e: ChangeEvent<HTMLInputElement>) => {
+    setSurName(e.target.value);
+    setSurNameValid(true);
+  };
+
+  const onChangePassword = (e: ChangeEvent<HTMLInputElement>) => {
+    setPassword(e.target.value);
+  };
+
+  const onValidatePassword = (progressScore: boolean) => {
+    setPasswordValid(progressScore);
+  };
+
+  const onBlurPassword = () => {
+    setIsPasswordErrorShow(true);
+  };
+
+  const onSubmit = async () => {
+    setIsLoading(true);
+    if (!name?.trim()) setNameValid(false);
+    if (!surName?.trim()) setSurNameValid(false);
+    if (!password.trim()) {
+      setPasswordValid(false);
+      setIsPasswordErrorShow(true);
+    }
+
+    if (!nameValid || !surNameValid || !password.trim() || !passwordValid) {
+      setIsLoading(false);
+      return;
+    }
+
+    const hash = createPasswordHash(password, passwordHash);
+
+    const loginData = {
+      userName: linkData.email ?? "",
+      passwordHash: hash,
+    };
+
+    const personalData = {
+      firstname: name,
+      lastname: surName,
+    };
+
+    try {
+      await activateConfirmUser({
+        personalData,
+        loginData,
+        key: linkData.confirmHeader ?? "",
+        userId: linkData.uid ?? "",
+        activationStatus: EmployeeActivationStatus.Activated,
+      });
+
+      setIsLoading(false);
+
+      window.location.replace(defaultPage);
+    } catch (error) {
+      const knownError = error as TError;
+      let errorMessage: string;
+
+      if (typeof knownError === "object") {
+        errorMessage =
+          knownError?.response?.data?.error?.message ||
+          knownError?.statusText ||
+          knownError?.message ||
+          "";
+      } else {
+        errorMessage = knownError;
+      }
+
+      setIsLoading(false);
+      toastr.error(errorMessage);
+    }
+  };
+
+  const activateConfirmUser = async ({
+    personalData,
+    loginData,
+    key,
+    userId,
+    activationStatus,
+  }: TActivateConfirmUser) => {
+    const changedData = {
+      id: userId,
+      FirstName: personalData.firstname,
+      LastName: personalData.lastname,
+    };
+
+    const { userName, passwordHash } = loginData;
+
+    const res1 = await changePassword(userId, passwordHash, key);
```

Comment:
try promise.all

Reply:
promise.all does not work because these requests must be executed one after another